api/ownership.go

```diff
@@ -182,7 +187,8 @@ func (o *Ownership) HasAnOwner() bool {
 
 // IsPublic returns true if there is no ownership in this resource
 func (o *Ownership) IsPublic() bool {
-	return !o.HasAnOwner()
+	return (o.Acls != nil && o.Acls.Public != nil &&
```

Comment:
Simplify the code by adding it as a function `IsAccessPermittedByPublic(accessType)`